src/js/flows/workspace/buildAndAuthenticateWorkspace.test.ts

```diff
@@ -0,0 +1,326 @@
+import {ipcRenderer, remote} from "electron"
+import jwtDecode from "jwt-decode"
+import Workspaces from "src/js/state/Workspaces"
+import WorkspaceStatuses from "src/js/state/WorkspaceStatuses"
+import {mocked} from "ts-jest/utils"
+import {createZealotMock} from "zealot"
+import Auth0Client from "../../auth0"
+import {AuthType} from "../../state/Workspaces/types"
+import initTestStore from "../../test/initTestStore"
+import {
+  buildAndAuthenticateWorkspace,
+  ConnectionError,
+  LoginError
+} from "./buildAndAuthenticateWorkspace"
+
+jest.mock("jwt-decode")
+jest.mock("../../auth0")
+jest.mock("electron", () => ({
+  ipcRenderer: {
+    invoke: jest.fn(),
+    once: jest.fn(),
+    removeListener: jest.fn()
+  },
+  remote: {
+    dialog: {
+      showMessageBox: jest.fn()
+    }
+  }
+}))
+
+const fxt = {
```

Comment:
Not sure what this is based on the name. Is this the mock zqd response?

Reply:
Oh sorry, it is just short for `fixtures` but since I use it so much I shortened the name. Shame on me

Reply:
Aww, maybe it means fixtures?